qiskit/circuit/quantumcircuit.py

```diff
@@ -206,6 +207,17 @@ def __init__(self, *regs, name=None, global_phase=0, metadata=None):
             raise TypeError("Only a dictionary or None is accepted for circuit metadata")
         self._metadata = metadata
 
+        self.qelib1_gate_names = ['barrier', 'measure', 'reset',
+                                  'u3', 'u2', 'u1', 'cx', 'id', 'u0',
+                                  'u', 'p', 'x', 'y', 'z', 'h', 's', 'sdg', 't', 'tdg',
+                                  'rx', 'ry', 'rz',
+                                  'sx', 'sxdg', 'cz', 'cy', 'swap', 'ch', 'ccx', 'cswap',
+                                  'crx', 'cry', 'crz', 'cu1', 'cp', 'cu3', 'csx', 'cu',
+                                  'rxx', 'rzz', 'rccx', 'rc3x', 'c3x', 'c3sx', 'c4x']
+        self.existing_composite_circuits = []
+        self.qasm_string = ""
+        self.qasm_string_temp = ""
```

Comment:
I'd prefer to keep the circuit class clean and not add new properties if possible (in particular not public ones):
* I think `self.qasm_string_temp` is used and not `self.qasm_string`, but it seems that even that variable could be kept locally instead of as a circuit property
* `self.existing_composite_circuits` exists only for caching reasons, right? I'm okay with that for now I think, but could you make it private?
* Do the qelib gates need to be a property or can they also be local?

Reply:
Let me check. I think they can be local variables. Sorry I am not an expert in python or programming in general. These might be silly questions:
- To make them local variables, should I just remove `self.` and use the variables name directly? 
- To make `self.existing_composite_circuits` "private, should I just add `_`, like `self._existing_composite_circuits`?
- I will remove `self.qasm_string` upon checking that it is not used.

Reply:
Yes to all of the above 😄